### PR TITLE
Fe feat/toggle save03

### PR DESCRIPTION
**Pull request by jmartin777**
# Description

ToggleSave Context Setup:

- created a ToggleSaveContext using createContext() to manage the toggle state.
-  defined the useToggleSave hook to access the context values.
- created a ToggleSaveProvider component to wrap your app with the context provider.

App Component:

- I implemented the handleToggleMushroom function to manage the state of the toggle for each mushroom card.
-  passed the isToggled and toggle props to the ToggleSave component when rendering mushroom cards.

DetailsPage Component:

- use the useToggleSave hook to access the toggle state and the toggle function from the context.
- passed the isToggled state to the ToggleSave component, ensuring that the toggle state persists between pages.

HomePage Component:

- used the useToggleSave hook to access the toggle state and the toggle function from the context.
- passed the isToggled state to the renderMushroomCards function when rendering mushroom cards.

MushroomCard Component:

-In the MushroomCard component, you added local state (isFav) to manage the favorite state of each card.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] Testing
- [x] Refactoring

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/ToggleSaveContext.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  const toggle = () => {
+    setIsToggled(!isToggled);
+  };
+
+  return (
+    <ToggleSaveContext.Provider value={{ isToggled, toggle }}>
+      {children}
+    </ToggleSaveContext.Provider>
+  );
```
Comment (Elise-Jones) [inline src/ToggleSaveContext.js:21]: Never new these method's existed and very cool to learn about them. Awesome work!

**Review by Elise-Jones — APPROVED** [review]

File: src/ToggleSaveContext.js (hunk 1 above)
Comment (jmartin777) [reply src/ToggleSaveContext.js:21]: Context.API

https://www.freecodecamp.org/news/context-api-in-react/

File: src/Components/App/App.js [hunk 2]
```diff
@@ -10,8 +10,8 @@ import MushroomCard from "../MushroomCard/MushroomCard";
 import DetailsPage from '../DetailsPage/DetailsPage';
 import ToggleSave from '../ToggleSave/ToggleSave';
 import { useQuery } from '@apollo/client';
-import GET_MUSHROOM_MATCHES from '../../queries.js'
-
+import { GET_MUSHROOM_MATCHES } from '../../queries';
+import { ToggleSaveProvider } from '../../ToggleSaveContext';
```
Comment (andreasorensen) [inline src/Components/App/App.js:14]: I'd love to hear more about this! I'm pumped we're implementing another library!

File: src/Components/App/App.js [hunk 3]
```diff
@@ -21,7 +21,9 @@ function App() {
     variables: { image: userImage },
     skip: !userImage,
   });
-  
+
+  const [mushroomStates, setMushroomStates] = useState({}); // State to track mushroom card states
```
Comment (andreasorensen) [inline src/Components/App/App.js:25]: I love this state functionality!

File: src/Components/App/App.js [hunk 4]
```diff
@@ -30,7 +32,14 @@ function App() {
     }
   }, [queryError])
 
-  const renderMushroomCards = () => {
+  const handleToggleMushroom = (mushroomId, isToggled) => {
+    setMushroomStates(prevState => ({
+      ...prevState,
+      [mushroomId]: isToggled,
+    }));
+  };
```
Comment (andreasorensen) [inline src/Components/App/App.js:40]: Very nice!

File: src/Components/App/App.js [hunk 5]
```diff
@@ -46,28 +55,36 @@ function App() {
             probability={mushroom.probability}
           />
         </NavLink>
-        <ToggleSave />
+        <ToggleSave
+          mushroomId={mushroom.id}
+          isSavedInitially={mushroomStates[mushroom.id] || false}
+          onToggle={handleToggleMushroom}
+          isToggled={isToggled} 
+          toggle={toggle}
```
Comment (andreasorensen) [inline src/Components/App/App.js:63]: Thanks for helping me understand this. I LOVE learning new tech!

File: src/Components/DetailsPage/DetailsPage.js [hunk 6]
```diff
@@ -3,59 +3,70 @@ import "./DetailsPage.css";
...
 
 const DetailsPage = () => {
-
   const location = useLocation();
   const mushroom = location.state;
   const taxonomy = mushroom.taxonomies[0];
   const characteristics = mushroom.characteristics[0];
   const lookAlikesValue = handleNoLookAlikes(mushroom.lookAlikes);
-
+  const { isToggled, toggle } = useToggleSave();
```
Comment (andreasorensen) [inline src/Components/DetailsPage/DetailsPage.js:14]: Muy interesante!

**Review by andreasorensen — APPROVED** [review]
This is amazing! If we have time we can figure out the persistence on the home page.

**Pull request merged**